quinn/src/endpoint.rs

```diff
@@ -68,6 +67,8 @@ impl Endpoint {
     /// client will only be able to connect to IPv6 servers. An IPv4 client is never dual-stack.
     #[cfg(feature = "ring")]
     pub fn client(addr: SocketAddr) -> io::Result<Self> {
+        use socket2::{Domain, Protocol, Socket, Type};
```

Comment:
I usually prefer to have guards on the imports -- better to keep this noise outside the main code and make the import blocks a little noisier IMO.